packages/keycloak-user-management/src/components/CreateEditUser/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { Col, notification, Row } from 'antd';
+import { RouteComponentProps } from 'react-router';
+import * as Yup from 'yup';
+import { Store } from 'redux';
+import { connect } from 'react-redux';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import { HeaderBreadCrumb } from '../HeaderBreadCrumb';
+import {
+  fetchKeycloakUsers,
+  KeycloakUser,
+  makeKeycloakUsersSelector,
+  reducer as keycloakUsersReducer,
+  reducerName as keycloakUsersReducerName,
+  getAccessToken,
+} from '@opensrp/store';
+import { KeycloakService } from '@opensrp/keycloak-service';
+import Ripple from '../Loading';
+import { UserForm, UserFormProps } from '../../forms';
+import '../../index.css';
+
+reducerRegistry.register(keycloakUsersReducerName, keycloakUsersReducer);
+
+/** inteface for route params */
+
+export interface RouteParams {
+  userId: string;
+}
+
+/** props for editing a user view */
+export interface Props {
+  accessToken: string;
+  fetchKeycloakUsersCreator: typeof fetchKeycloakUsers;
+  keycloakUser: KeycloakUser | null;
+  serviceClass: typeof KeycloakService;
+}
+
+/** type intersection for all types that pertain to the props */
+export type PropsTypes = Props & RouteComponentProps<RouteParams>;
+
+/** default form initial values */
+
+export const defaultInitialValues: KeycloakUser = {
+  access: {
+    manageGroupMembership: false,
+    view: false,
+    mapRoles: false,
+    impersonate: false,
+    manage: false,
+  },
+  createdTimestamp: undefined,
+  disableableCredentialTypes: [],
+  email: '',
+  emailVerified: false,
+  enabled: true,
+  firstName: '',
+  id: '',
+  lastName: '',
+  notBefore: 0,
+  requiredActions: [],
+  totp: false,
+  username: '',
+};
+
+/** default props for editing user component */
+export const defaultProps: Partial<PropsTypes> = {
```

Comment:
Is it necessary to have this as Partial?  I'm always wary of using Partial because it weakens the type def slightly

Reply:
@kelvin-muchiri was this one addressed?